src/interface/src/app/signup/signup.component.ts

```diff
@@ -52,25 +53,36 @@ export class SignupComponent {
   signup() {
     if (this.submitted) return;
 
+    this.submitted = true;
```

Comment:
Two suggestions here, maybe this should be called `submitting` and you should probably use this value to disable the submit button on the template